phoenix-core/src/main/java/org/apache/phoenix/schema/json/PhoenixJson.java

```diff
@@ -0,0 +1,280 @@
+                throw new PhoenixJsonException("path: " + path + " not found");
+            }
+            node = nodeTemp;
+        }
+        return new PhoenixJson(node);
+    }
+
+    /**
+     * Get {@link PhoenixJson} for a given json paths. For example :
+     * <p>
+     * <code>
+     * {"f2":{"f3":1},"f4":{"f5":99,"f6":{"f7":"2"}}}'
+     * </code>
+     * <p>
+     * for this source json, if we want to know the json at path {'f4','f6'} it will return
+     * {@link PhoenixJson} object for json {"f7":"2"}. It always returns the last key if same key
+     * exist more than once.
+     * <p>
+     * If the given path is unreachable then it return null.
+     * @param paths {@link String []} of path in the same order as they appear in json.
+     * @return {@link PhoenixJson} for the json against @paths.
+     */
+    public PhoenixJson getNullablePhoenixJson(String[] paths) {
+        JsonNode node = this.node;
+        for (String path : paths) {
+            JsonNode nodeTemp = null;
+            if (node.isArray()) {
+                try {
+                    int index = Integer.parseInt(path);
+                    nodeTemp = node.path(index);
+                } catch (NumberFormatException nfe) {
+                    return null;
+                }
+            } else {
+                nodeTemp = node.path(path);
+            }
+            if (nodeTemp == null || nodeTemp.isMissingNode()) {
+                return null;
+            }
+            node = nodeTemp;
+        }
+        return new PhoenixJson(node);
+    }
+
+    /**
+     * Serialize the current {@link PhoenixJson} to String. Its required for
+     * json_extract_path_text(). If we just return node.toString() it will wrap String value in
+     * double quote which is not the expectation, hence avoiding calling toString() on
+     * {@link JsonNode} until PhoenixJson represent a Json Array or container for Json object. If
+     * PhoenixJson just represent a {@link ValueNode} then it should return value returned from
+     * objects toString().
+     */
+    public String serializeToString() {
+        if (this.node == null || this.node.isNull()) {
+            return null;
+        } else if (this.node.isValueNode()) {
+
+            if (this.node.isNumber()) {
+                return this.node.getNumberValue().toString();
+            } else if (this.node.isBoolean()) {
+                return String.valueOf(this.node.getBooleanValue());
+            } else if (this.node.isTextual()) {
+                return this.node.getTextValue();
+            } else {
+                return getJsonAsString();
+            }
+        } else if (this.node.isArray()) {
+            return getJsonAsString();
+        } else if (this.node.isContainerNode()) {
+            return getJsonAsString();
+        }
+
+        return null;
+
+    }
+
+    @Override
+    public String toString() {
+        if (this.jsonAsString == null && this.node != null) {
```

Comment:
can jsonAsString ever be null, seems like you create a PhoenixJson object you it will always have be from a string